var/spack/repos/builtin/packages/paraview/package.py

```diff
@@ -226,6 +223,9 @@ class Paraview(CMakePackage, CudaPackage, ROCmPackage):
     depends_on("protobuf@3.4:3.18", when="@:5.10%intel@2021:")
     depends_on("protobuf@3.4:3.18", when="@:5.10%xl")
     depends_on("protobuf@3.4:3.18", when="@:5.10%xl_r")
+    # protobuf requires newer abseil-cpp, which in turn requires C++14,
+    # but paraview uses C++11 by default
+    depends_on("protobuf@3.4:3.21", when="@:5.11")
```

Comment:
I don't see how this helps, protobuf doesn't put an upperbound on abseil to the versions you've constrained it to here.

Reply:
It puts a _lower_ bound:

https://github.com/spack/spack/blob/6576655137b847af102ebcea20278a6d4f9f09e7/var/spack/repos/builtin/packages/protobuf/package.py#L86

Before that it doesn't depend on abseil.  So the:
```python
# Newer abseil-cpp requires C++14, but paraview uses C++11 by default
conflicts("^abseil-cpp@2023:")
```
is really an indirect way of selecting a `protobuf` that is old enough to not use abseil. Since `protobuf` is a declared dependency, and `abseil-cpp` is not, I moved the constraint directly to `protobuf`.

Reply:
`conflicts("^abseil-cpp", when="^abseil-cpp")` ;p

Reply:
Feels awkward to duplicate those bounds though.